src/rws_interface.cpp

```diff
@@ -81,6 +82,27 @@ RWSInterface::StaticInfo RWSInterface::collectStaticInfo()
   return static_info;
 }
 
+std::vector<RWSInterface::RobotWareOptionInfo> RWSInterface::getPresentRobotWareOptions()
+{
+  std::vector<RobotWareOptionInfo> result;
+
+  RWSClient::RWSResult rws_result = rws_client_.getConfigurationInstances(Identifiers::SYS,
+                                                                          Identifiers::PRESENT_OPTIONS);
+
+  std::vector<Poco::XML::Node*> node_list = xmlFindNodes(rws_result.p_xml_document, XMLAttributes::CLASS_CFG_IA_T_LI);
+
+  for (size_t i = 0; i < node_list.size(); i+=2)
```

Comment:
Just noticed this: `i+2`.

Should we make sure that `node_list` actually contains a multiple of 2 items?

Reply:
There's an if case inside the for loop for that, but what I really should do (and I was thinking about it) is to implement a `xmlFindChildNodes(...)` function to get a list of a XML node's the children that for example has a specific XML attribute.